

## Pipeline usage

### Environment preparation
you can install either install only `torch==1.12` by pip or use conda for full environment clone.

```bash
conda env create -f env.yml
conda activate motifNet

python test_env.py

```

`test_env.py` will tell whether your environment has been installed sucessfully.

### Data Preparation
Prepare your RNA sequencing results table file with `delta_psi', 'p_value', 'q_value', 'read_counts' columns.

```
python preprocess.py -i /Path/to/your/folder/Branaplam-DMSO_donors.mntjulip.txt -o data/train.csv --seq_len 256
```

### Train

```bash
python main.py train -i data/train.csv --seq_len 256 --prefix RC00106


"""
	-i train data table 
	--seq_len input sequence length of train.csv 
	-v validation data table [optional]  
	--prefix output prefix for model files and logs
"""
```

If you want to split train and validation, for example by leaving some chromosomes for validation.
Simply modify preprocess.py to your needs and provide an extra `-v test.csv`.


## Data visualization
After the model was trained, ranked motif logos plot and PCA of splicing junction sequence feature can be generated by script `draw_results`.

```bash
python draw_results.py -i data/train.csv --prefix results/RC01 --ckpt ckpt/e5.pth --seq_len 256 --psi_threshold 0.2
```

### ranked motif logos
![motif positional importance](imgs/motif.png)

### PCA plot
![PCA of motifs nearby splicing junctions among genes](imgs/pca.png)


### inference 
If you just want to look into some some specific genes sequences, you can use the following pipeline to generate predict delta psi and motif fingerprint.

```bash
python predict_seq.py -i data/SMN_seq.csv --prefix results/SMN
```


```python
fingerprint = np.load('results/output.npy')
print(fingerprint.shape)

Out[4]: (10, 256, 32)
```
where 10 is number of rows in seq.csv, 256 is the length of sequences and 32 the number of motifs.

